include/seastar/core/future.hh

```diff
@@ -234,8 +234,9 @@ struct future_state_base {
     enum class state : uintptr_t {
          invalid = 0,
          future = 1,
-         result = 2,
-         exception_min = 3,  // or anything greater
+         result_unavailable = 2,
+         result = 3,
```

Comment:
`future::available()` (which returns `true` when `_state` already stores result/value or error) still could be implemented  with single comparison only.